js/background.js

```diff
@@ -568,6 +568,47 @@
       }
     });
 
+    Whisper.events.on('onEditProfile', () => {
+      const ourNumber = textsecure.storage.user.getNumber();
+      const profile = storage.getLocalProfile();
+      const displayName = profile && profile.name && profile.name.displayName;
+      if (appView) {
+        appView.showNicknameDialog({
+          title: 'Change your own display name',
+          message: 'Note: Your display name will be visible to your contacts.',
+          nickname: displayName,
+          onOk: async (newName) => {
+            // Update our profiles accordingly'
+            const trimmed = newName && newName.trim();
+            let newProfile = null;
+            if (_.isEmpty(trimmed)) {
+              await storage.removeLocalProfile();
```

Comment:
Is this function overwriting the profile every time? Maybe should just update the name field to avoid problems if we decide to store other data in the profile?